### PR TITLE
Move to lib

### PR DESCRIPTION
**Pull request by michaelmcinerney**

### CODE REVIEW

**Review by lsf37 — APPROVED** [review]
:+1:

File: proof/refine/ARM/Schedule_R.thy [hunk 1]
```diff
@@ -2303,19 +2303,6 @@ lemma setReprogramTimer_active_sc_at'[wp]:
   unfolding active_sc_at'_def
   by wpsimp
 
-(* FIXME RT: move these whileM lemmas, prove `whileM_*_inv` using `whileM_wp_gen`. *)
```
Comment (corlewis) [inline proof/refine/ARM/Schedule_R.thy:2306]: Very minor, but do we want to keep the second half of this FIXME? I have to admit, that it's not completely clear to me what the intention of it was, so maybe we don't care about it anymore

**Review by corlewis — APPROVED** [review]
Thanks for cleaning these up!

File: proof/refine/ARM/Schedule_R.thy (hunk 1 above)
Comment (michaelmcinerney) [reply proof/refine/ARM/Schedule_R.thy:2306]: This turned out to be easy. The tests failed for some probably annoying reason, so we may as well update this and rerun the tests

File: lib/LemmaBucket.thy [hunk 2]
```diff
@@ -561,4 +561,28 @@ lemma cases_conj_strg: "A \<and> B \<longrightarrow> (P \<and> A) \<or> (\<not>
...
+  apply (case_tac list; simp)
+  apply (fastforce simp: filter_id_conv)
+  done
+
+lemma case_list_when:
+  "(case l of
+      [] \<Rightarrow> return ()
+      | r # x \<Rightarrow> f r x)
+   = (when (l \<noteq> []) $ f (hd l) (tl l))"
+  by (clarsimp simp: list_case_If2)
```
Comment (Xaphiosis) [inline lib/LemmaBucket.thy:586]: since we're moving these, could we also clean up style? I don't think this one follows any

**Comment by Xaphiosis** [comment]
Had one comment, not super relevant, otherwise looks good. Thanks for doing cleanup.

**Review by Xaphiosis — APPROVED** [review]

File: lib/LemmaBucket.thy (hunk 2 above)
Comment (michaelmcinerney) [reply lib/LemmaBucket.thy:586]: Is it that the `[]` and the `r` should be indented the same amount? That's the only thing I can see here

Comment (lsf37) [reply lib/LemmaBucket.thy:586]: The `|` should align with `case`. This this case it looks like the whole case expression would fit on one line, which I would usually prefer. Also the naming convention for lists is with `s`, so it should be `r # xs`

Comment (michaelmcinerney) [reply lib/LemmaBucket.thy:583]: I put it all on the one line, and update the name of the variable

**Pull request merged**